src/SourceBuild/content/eng/tools/BuildComparer/Program.cs

```diff
@@ -532,15 +553,164 @@ public async Task EvaluateBlobContents(AssetMapping mapping)
+                            if (nextBaseEntry != null)
+                            {
+                                CompareAvailableStreams(mapping, baselineStreams, diffStreams, nextBaseEntry);
+                            }
+
+                            nextDiffEntry = await WalkNextCommon(commonFiles, diffReader, diffStreams);
+                            if (nextDiffEntry != null)
+                            {
+                                CompareAvailableStreams(mapping, baselineStreams, diffStreams, nextDiffEntry);
+                            }
+                        }
+                        while (nextBaseEntry != null || nextDiffEntry != null);
+
+                        // If there are any remaining streams, create an evaluation error
+                        if (baselineStreams.Count > 0 || diffStreams.Count > 0)
+                        {
+                            mapping.EvaluationErrors.Add("Failed to compare all tar entries.");
+                        }
+                    }
+                }
+            }
+        }
+
+        // Walk the tar to the next entry that exists in both the base and the diff
+        static async Task<string> WalkNextCommon(HashSet<string> commonFiles, TarReader reader, Dictionary<string, Stream> streams)
+        {
+            TarEntry baseEntry;
+            while ((baseEntry = reader.GetNextEntry()) != null && baseEntry.DataStream != null)
+            {
+                string entryStripped = RemoveVersionsNormalized(baseEntry.Name);
+                // If the element lives in the common files hash set, then copy it to a memory stream.
+                // Do not close the stream.
+                if (commonFiles.Contains(entryStripped))
+                {
+                    streams[entryStripped] = await CopyStreamToSeekableStreamAsync(baseEntry.DataStream);
+                    return entryStripped;
+                }
+            }
+            return null;
+        }
+
+        // Given we have a new entry that is common between base and diff, attempt to do some comparisons.
+        void CompareAvailableStreams(AssetMapping mapping, Dictionary<string, Stream> baselineStreams, Dictionary<string, Stream> diffStreams,
+             string entry)
+        {
+            if (baselineStreams.TryGetValue(entry, out var baselineFileStream) &&
+                diffStreams.TryGetValue(entry, out var diffFileStream))
+            {
+                CompareAssemblyVersions(mapping, entry, baselineFileStream, diffFileStream);
+                baselineFileStream.Dispose();
+                diffFileStream.Dispose();
+                baselineStreams.Remove(entry);
+                diffStreams.Remove(entry);
+            }
+        }
+    }
+
+    private static string RemoveVersionsNormalized(string path)
+    {
+        string strippedPath = path.Replace("\\", "//");
+        string prevPath = path;
```

Comment:
nit: this is overwritten on the next line anyway so initializing to `path` is unnecessary